default_context.go

```diff
@@ -217,12 +219,12 @@ func (d *DefaultContext) Redirect(status int, url string, args ...interface{}) e
 
 // Data contains all the values set through Get/Set.
 func (d *DefaultContext) Data() map[string]interface{} {
+	d.moot.Lock()
```

Comment:
Since you protected `Set` for write, can't read still use a RLock?

Reply:
I upgraded the mutex just to be careful and safe. I think it’s worth it.